### PR TITLE
Feature pubchat v2

### PR DESCRIPTION
**Pull request by nottoast**

### CODE REVIEW

**Comment by matkastner** [comment]
deploying to beta for testing

File: app/google-services.json [hunk 1]
```diff
@@ -52,6 +60,13 @@
           "status": 2,
           "other_platform_oauth_client": [
             {
+              "client_id": "913036450395-tdr4ckbc7aep1sdbsqh1opc8qj1ptcfj.apps.googleusercontent.com",
+              "client_type": 2,
+              "ios_info": {
+                "bundle_id": "com.example.Pub-Thursday"
```
Comment (matkastner) [inline app/google-services.json]: this package name is wrong, is this change doing anything?

File: app/src/main/java/com/pubbyt/pubthursdayleaderboard/MainActivity.java [hunk 2]
```diff
@@ -365,6 +396,18 @@ public boolean onOptionsItemSelected(MenuItem item) {
     @Override
     public boolean onNavigationItemSelected(MenuItem item) {
         Log.d(TAG, "onNavigationItemSelected: " + item);
+
+        switch(item.getItemId()) {
+            case R.id.nav_pub_chat:
+                Intent messengerIntent = new Intent(this, MessengerActivity.class);
+                startActivity(messengerIntent);
+                this.overridePendingTransition(0, 0);
```
Comment (matkastner) [inline app/src/main/java/com/pubbyt/pubthursdayleaderboard/MainActivity.java]: this kinda makes the transition weird on my phone, I think it is better without personally

File: app/src/main/java/com/pubbyt/pubthursdayleaderboard/MessengerActivity.java [hunk 3]
```diff
@@ -0,0 +1,562 @@
...
+
+    public ArrayList<Message> messages;
+
+    private static String messageToSend = "";
+    private static String messageType = "";
+    private static String colour = "";
+    private static String userNickname = "";
+    private static Long lastNotificationSent = 0L;
+
+    public ProgressBar progress_bar;
```
Comment (matkastner) [inline app/src/main/java/com/pubbyt/pubthursdayleaderboard/MessengerActivity.java]: why aren't these camelCase?

Comment (matkastner) [inline app/src/main/java/com/pubbyt/pubthursdayleaderboard/MessengerActivity.java]: https://source.android.com/source/code-style.html

File: app/src/main/res/values/styles.xml [hunk 4]
```diff
@@ -20,4 +20,13 @@
     <style name="LeaderboardListStyle" parent="TextAppearance.AppCompat.Medium">
         <item name="android:textColor">#000000</item>
     </style>
+
+    <style name="AppTheme2" parent="Theme.AppCompat.Light.DarkActionBar">
```
Comment (matkastner) [inline app/src/main/res/values/styles.xml]: could this be like pubChatTheme or something more descriptive?

File: app/src/main/java/com/pubbyt/pubthursdayleaderboard/MessengerActivity.java [hunk 5]
```diff
@@ -0,0 +1,570 @@
...
+    public static String userName = PersonStore.getCurrentUserName();
+    public static String messageColour = DEFAULT_COLOUR;
+    public static String emojiButton = DEFAULT_EMOJI;
+    public static Long lastMessageTime = 0L;
+    public static Long lastUserTime = 0L;
+    public static int loadedHistory = DEFAULT_HISTORY;
+    public static int currentHistory = DEFAULT_HISTORY;
+
+    public final FirebaseDatabase database = FirebaseDatabase.getInstance();
+    public DatabaseReference messagesTable;
```
Comment (matkastner) [inline app/src/main/java/com/pubbyt/pubthursdayleaderboard/MessengerActivity.java]: feel like this should be private and shouldn't be part of activity - should be a service or singleton?

File: app/src/main/res/values/styles.xml (hunk 4 above)
Comment (nottoast) [inline app/src/main/res/values/styles.xml]: Done

File: app/src/main/java/com/pubbyt/pubthursdayleaderboard/MainActivity.java (hunk 2 above)
Comment (nottoast) [inline app/src/main/java/com/pubbyt/pubthursdayleaderboard/MainActivity.java]: Done

File: app/google-services.json (hunk 1 above)
Comment (nottoast) [inline app/google-services.json]: Removed

**Comment by nottoast** [comment]
all updated mate, please merge to developer :p

**Pull request merged**